### PR TITLE
feat: add inside pages component

### LINKED ISSUE #19: [DRAFT] Redesign UiMegaMenu

### PR DESCRIPTION
**Pull request by pspaczek**
# Related issue
Close #19 

## Scope of work
- create UiContentArea (it's the best name for it?) component.
- support multi-level UiContentArea 
## Things to do

- [x]  support multilevel UiContentArea 
- [x] add stories
- [x] add unit tests

## Screenshots of visual changes
Kapture 2022-11-02 at 12 52 33

**Multi-level**
Kapture 2022-11-02 at 12 53 54

**Mobile Menu Example**
Kapture 2022-11-02 at 12 57 03

### CODE REVIEW

**Comment by AlanLes** [comment]
Regarding the name, imo `UiContentArea` doesn't say much about this component. Almost everything has content, and everywhere is an area for that content 😅  When I look at this, I see something that could be named like `UiHorizontalPaging`, wdyt @pspaczek ? 😄

File: src/components/organisms/UiContentArea/UiContentArea.spec.ts [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+  it('component allow to pass items by default slot', async () => {
+    const wrapper = mount(UiContentArea, {
+      props: { title: 'Settings & Info' },
+      slots: { default: h(UiContentAreaItem, item) },
+    });
+    await nextTick();
+    const menuItem = wrapper.findComponent(UiMenuItem);
+    expect(menuItem.html()).toContain(item.label);
+  });
+  it('component allot to pass items by items prop', async () => {
```
Comment (AlanLes) [inline src/components/organisms/UiContentArea/UiContentArea.spec.ts:35]: `allows`; typo 🙂

File: src/components/organisms/UiContentArea/UiContentArea.spec.ts [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+        modelValue: item,
+        title: 'Settings & Info',
+        items: [ item ],
+      },
+    });
+    await nextTick();
+    const backButton = wrapper.find('.ui-content-area__back');
+    expect(backButton.exists()).toBe(true);
+  });
+  it('component support nesting', async () => {
```
Comment (AlanLes) [inline src/components/organisms/UiContentArea/UiContentArea.spec.ts:58]: it `supports`, not `support`, typo 🙂

File: src/components/organisms/UiContentArea/UiContentArea.spec.ts [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+        title: 'Settings & Info',
+        items: [ item ],
+      },
+      slots: { [item.name]: h(nestedComponent) },
+    });
+    await nextTick();
+    const nestedContent = wrapper.find(`[data-testid="${nestedItem.name}"]`);
+    expect(nestedContent.exists()).toBe(true);
+  });
+  it('back button in component with nesting remove last active item', async () => {
```
Comment (AlanLes) [inline src/components/organisms/UiContentArea/UiContentArea.spec.ts:78]: `removes`

File: src/components/organisms/UiContentArea/UiContentArea.spec.ts [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+  title: 'Medical certification and compliance',
+  name: 'medical-certification',
+};
+
+describe('UiContentArea.vue', () => {
+  it('renders a component', () => {
+    const wrapper = mount(UiContentArea);
+    expect(wrapper.classes('ui-content-area')).toBe(true);
+  });
+  it('component allow to pass items by default slot', async () => {
```
Comment (AlanLes) [inline src/components/organisms/UiContentArea/UiContentArea.spec.ts:26]: `allows`; typo 🙂

File: src/components/organisms/UiContentArea/UiContentArea.vue [hunk 5]
```diff
@@ -0,0 +1,293 @@
...
+import UiContentAreaItem from './_internal/UiContentAreaItem.vue';
+
+export interface ContentAreaItem {
+  label: string;
+  title: string;
+  name?: string;
+  [key: string]: unknown;
+}
+
+const props = defineProps({
```
Comment (AlanLes) [inline src/components/organisms/UiContentArea/UiContentArea.vue:117]: I wrote it already in [this PR](https://github.com/infermedica/component-library/pull/43), so you can dig into the discussion; 
**tldr** Why not use `withDefaults` instead of casting those props types over and over? 😜

File: src/components/organisms/UiContentArea/UiContentArea.vue [hunk 6]
```diff
@@ -0,0 +1,293 @@
...
+}
+));
+
+const emit = defineEmits<{(event: 'update:modelValue', value: ContentAreaItem[]): void;
+}>();
+
+const index = inject('index', 0);
+provide('index', index + 1);
+const isNested = computed(() => index > 0);
+const headerIsDisplayed = computed(() => (props.hasHeader && !isNested.value));
```
Comment (AlanLes) [inline src/components/organisms/UiContentArea/UiContentArea.vue:193]: I don't know about your habits, but I see other `booleans` variables are named with regarding the _**Answering to the question**_ pattern, so WDYT about renaming this `headerIsDisplayed` to `isHeaderDisplayed`? :)

File: src/components/organisms/UiContentArea/UiContentArea.vue [hunk 7]
```diff
@@ -0,0 +1,293 @@
...
+provide('activeItems', activeItems);
+const sizeOfActiveItems = computed(() => (activeItems.value.length));
+const activeItem = computed<ContentAreaItem>(() => (activeItems.value[index] || {}));
+const isActive = computed(() => (Object.keys(activeItem.value).length > 0));
+const activeItemName = computed(() => activeItem.value?.name || '');
+provide('activeItemName', activeItemName);
+const currentTitle = computed(() => (activeItems.value[sizeOfActiveItems.value - 1]?.title || props.title));
+const backToTitle = computed(() => (activeItems.value[sizeOfActiveItems.value - 2]?.title || props.title));
+
+const itemsToHandle = ref({});
```
Comment (AlanLes) [inline src/components/organisms/UiContentArea/UiContentArea.vue:210]: maybe `ref<ContentAreaItem>({})`; or any other proper type? 🙂

File: src/components/organisms/UiContentArea/UiContentArea.vue [hunk 8]
```diff
@@ -0,0 +1,293 @@
...
+provide('items', itemsToHandle);
+const itemsAsArray = computed<ContentAreaItem[]>(() => (Object.values(itemsToHandle.value)));
+const menuItems = computed(() => {
```
Comment (AlanLes) [inline src/components/organisms/UiContentArea/UiContentArea.vue:213]: Are you able to provide the type for this `computed` property, as you did on the line before? :)

File: src/components/organisms/UiContentArea/_internal/UiContentAreaItem.vue [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+const activeItemName = inject('activeItemName') as Ref<string>;
+const isActive = computed(() => activeItemName.value === props.name);
+
+const item = computed(() => ({
+  label: props.label,
+  title: props.title,
+  name: props.name,
+}));
+const items = inject('items') as Ref<Record<string, ContentAreaItem>>;
+items.value[props.name] = item.value;
```
Comment (AlanLes) [inline src/components/organisms/UiContentArea/_internal/UiContentAreaItem.vue:47]: Why does it happen in the global scope of the script? 🤔 🙂

**Review by AlanLes — APPROVED** [review]
Really great! Awesome component! 🤩 ⭐

File: src/components/organisms/UiContentArea/UiContentArea.vue (hunk 5 above)
Comment (pspaczek) [reply src/components/organisms/UiContentArea/UiContentArea.vue]: This way, unsupport docs comment and disallow to add props descriptions to the Storybook, but @ArkadiuszKrzymienski will change that.

File: src/components/organisms/UiContentArea/_internal/UiContentAreaItem.vue (hunk 9 above)
Comment (pspaczek) [reply src/components/organisms/UiContentArea/_internal/UiContentAreaItem.vue]: global scope? It's use `itemsToHandle` in UiHorizontalPaging. It's required to support adding items by default slot. Item add it to `itemsToHandle` and it will be displayed in menu

**Comment by pspaczek** [comment]
:thank_you: @AlanLes for yours review
